emsdk.py

```diff
@@ -1241,13 +1241,32 @@ def emscripten_npm_install(tool, directory):
   print('Running post-install step: npm ci ...')
   try:
     subprocess.check_output(
-        [npm, 'ci', '--production'],
+        [npm, 'ci', '--production', '--no-optional'],
         cwd=directory, stderr=subprocess.STDOUT, env=env,
         universal_newlines=True)
   except subprocess.CalledProcessError as e:
     print('Error running %s:\n%s' % (e.cmd, e.output))
     return False
 
+  # Manually install the appropriate native Closure Compiler package
```

Comment:
Could you leave a longer comment here about what this is needed?

Can you mention the upstream closure bug that bloats the native package: https://github.com/google/closure-compiler-npm/issues/186

And also mention that its only need because we use `npm ci` (which seems to install *all* optional dependencies) as opposed to `npm install` which does the right thing and installs just the current platform one :(

I'm hoping we might be able to revert this if they fix either or both of these issues.

Reply:
How's this look?